File: VB/CustomWizardExample/Person.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Xml.Linq

Namespace CustomWizardExample
    Public Class Person
        <DisplayName("First Name")> _
        Public Property FirstName() As String
        <DisplayName("Second Name")> _
        Public Property SecondName() As String
    End Class
    Public Class PersonsSource
        Public Function GetXmlPersons() As IList(Of Person)
            Dim root = System.Xml.Linq.XDocument.Load("Data.xml").Root
            Dim persons As New List(Of Person)()
            For Each item In root.Elements()
                Dim firstName = GetAttributeValue(item, "FirstName")
                Dim secondName = GetAttributeValue(item, "SecondName")

                Dim person = New Person() With { _
                    .FirstName = firstName, _
                    .SecondName = secondName _
                }
                persons.Add(person)
            Next item
            Return persons
        End Function
        Private Shared Function GetAttributeValue(ByVal element As XElement, ByVal attributeName As String) As String
            Dim attrbute As XAttribute = element.Attribute(attributeName)
            Return If(attrbute Is Nothing, Nothing, attrbute.Value)
        End Function
        Public Function GetStaticPersons() As IList(Of Person)
            Return New List(Of Person)( { _
                New Person() With { _
                    .FirstName = "John", _
                    .SecondName = "Abbot" _
                }, _
                New Person() With { _
                    .FirstName = "Paul", _
                    .SecondName = "Bass" _
                }, _
                New Person() With { _
                    .FirstName = "George", _
                    .SecondName = "Chance" _
                } _
            })
        End Function
    End Class
End Namespace

File: CS/CustomWizardExample/Person.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace CustomWizardExample {
    public class Person {
        [DisplayName("First Name")]
        public string FirstName { get; set; }
        [DisplayName("Second Name")]
        public string SecondName { get; set; }
    }
    public class PersonsSource {
        public IList<Person> GetXmlPersons() {
            var root = System.Xml.Linq.XDocument.Load("Data.xml").Root;
            List<Person> persons = new List<Person>();
            foreach(var item in root.Elements()) {
                var firstName = GetAttributeValue(item, "FirstName");
                var secondName = GetAttributeValue(item, "SecondName");

                var person = new Person() { FirstName = firstName, SecondName = secondName };
                persons.Add(person);
            }
            return persons;
        }
        static string GetAttributeValue(XElement element, string attributeName) {
            XAttribute attrbute = element.Attribute(attributeName);
            return attrbute == null ? null : attrbute.Value;
        }
        public IList<Person> GetStaticPersons() {
            return new List<Person>(new[] { 
                        new Person() { FirstName = "John", SecondName = "Abbot" }, 
                        new Person() { FirstName = "Paul", SecondName = "Bass" },
                        new Person() { FirstName = "George", SecondName = "Chance" }});
        }
    }
}
